src/ol/renderer/canvas/canvasvectorlayerrenderer.js

```diff
@@ -194,6 +227,14 @@ ol.renderer.canvas.VectorLayer.prototype.prepareFrame =
 
   var extent = ol.extent.buffer(frameStateExtent,
       vectorLayerRenderBuffer * resolution);
+  var projectionExtent = viewState.projection.getExtent();
+
+  if (vectorSource.getWrapX() && viewState.projection.isGlobal() &&
```

Comment:
Looking at our examples, the use case of just rendering vector data is more frequent than editing it, so wrapX defaulting to true makes sense. Not sure if our examples are representative for real world use though...